hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/AvroConversionHelper.scala

```diff
@@ -343,25 +350,67 @@ object AvroConversionHelper {
       case structType: StructType =>
         val schema: Schema = convertStructTypeToAvroSchema(structType, structName, recordNamespace)
         val childNameSpace = if (recordNamespace != "") s"$recordNamespace.$structName" else structName
-        val fieldConverters = structType.fields.map(field =>
-          createConverterToAvro(
-            field.dataType,
-            field.name,
-            childNameSpace))
-        (item: Any) => {
-          if (item == null) {
-            null
-          } else {
-            val record = new Record(schema)
-            val convertersIterator = fieldConverters.iterator
-            val fieldNamesIterator = dataType.asInstanceOf[StructType].fieldNames.iterator
-            val rowIterator = item.asInstanceOf[Row].toSeq.iterator
+        if (honorSchemaMismatch) {
```

Comment:
there are few lines that could be common among existing code flow and new code flow in this code block of interest. but I did not want to do if else line by line. and hence a big if else for old code path and new code path.

Reply:
ok

Reply:
can you add some context on what exactly is the fix here?